### ISSUE TITLE
after updating 5.11 to 5.11.4 having an error in "acf_get_value was called incorrectly"

### ISSUE THREAD

**Issue by incohe** [AdvancedCustomFields/acf#591]
when updating ACF Pro 5.11 to 5.11.4 or 5.11.3 im having error in acf_get_value function thats makes my website white page. please fix this asap

thanks

wp debug error:
PHP Notice:  acf_get_value was called <strong>incorrectly</strong>. <strong>Advanced Custom Fields PRO</strong> - We've detected one or more calls to retrieve ACF field values before ACF has been initialized. This is not supported and can result in malformed or missing data. <a href="https://www.advancedcustomfields.com/resources/acf-field-functions/" target="_blank">Learn how to fix this</a>. Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 5.11.1.) in C:\xampp_new\htdocs\[website-folder]\wp-includes\functions.php on line 5663
![web page error](https://user-images.githubusercontent.com/55270275/146712058-8d1953fc-a685-464e-8286-4d075498e316.PNG)
![after update error](https://user-images.githubusercontent.com/55270275/146712112-7c3f3a11-4c31-456a-923c-5208057641e7.PNG)

**Comment by lgladdy** [comment]
@incohe Hey, if you're getting this error, it means you're trying to call ACF functions too early - before it's been initialised and knows what fields are available.

The links in the error should give you more information on what's going wrong and how to fix it.

This error itself won't be causing the 500 errors, but it's possible if you're trying to load non-ACF option values with get_field, they could be returning null now, which depending on the coding could cause this.

**Comment by incohe** [comment]
> @incohe Hey, if you're getting this error, it means you're trying to call ACF functions too early - before it's been initialised and knows what fields are available.
> 
> The links in the error should give you more information on what's going wrong and how to fix it.
> 
> This error itself won't be causing the 500 errors, but it's possible if you're trying to load non-ACF option values with get_field, they could be returning null now, which depending on the coding could cause this.

i only have error when updating the acf to latest version or maybe because of your explanation i do call get_field field before wp_head/get_header function but other than that i dont have other modification in the plugin im still puzzled because it never happened to be before i start using acf since version 3

**Comment by lgladdy** [comment]
@incohe Hey, yeh - changes made in ACF 5.11 will have caused this error - but unfortunately the changes were required for security purposes. Calling before wp_head or get_header isn't necessarily an issue - but you need to do wrap any code that is executed as loaded by WordPress (such as functions.php, or a plugin) in an action hook on `acf/init`.

We plan to make further changes in a future version to log exactly where this issue is happening, but for now, if you add an error handler to print a backtrace when a notice is fired - you'll be able to find the line of code that's at fault. Something like:

```php
set_error_handler(function() {
    error_log(print_r(debug_backtrace(), true));
    return true;
}, E_USER_NOTICE);
```

should work!

**Comment by incohe** [comment]
i check my init i have only one that have get_field in init event i remove
it or change it to acf/init i still have error and having memory issue
[image: image.png]

**Comment by incohe** [comment]
Upon checking i found that  the error i have i only on the v2 theme that i
make and one of the utility function such a troublesome
[image: image.png]

Thanks you for your patients now i can upgrade the plugin w/ not errors
maybe this is also the one slowing down may page edit loading
thanks

**Comment by lgladdy** [comment]
@incohe Good to hear it's sorted! The images don't show when sent by email, so feel free to upload them manually on this issue if you need more help figuring out exactly what's gone wrong! :)

**Comment by incohe** [comment]
apologies the issue is i use locale filter
![image](https://user-images.githubusercontent.com/55270275/146938259-18323014-ae63-414b-b22a-3037e9ae0385.png)
i just convert it to 
![image](https://user-images.githubusercontent.com/55270275/146938325-759afe97-31a5-4110-8bc9-831d187e6fc7.png)
after that i don't have issue but upon checking i have error on acf js in option page but other page seems ok
![image](https://user-images.githubusercontent.com/55270275/146938607-ea699c7f-c3b8-4b07-943d-eceef813bfd3.png)
ill just wait for the next version i hope the js error will be fixed on the old version 5.11 i dont have issue with it 

Thanks
